ledger/ledger_test.go

```diff
@@ -0,0 +1,308 @@
+		},
+		expected: `
+Date       | Description               | Change
+01/01/2015 | Buy present               |      (€10.00)
+`,
+	},
+	{
+		name:     "Dutch locale",
+		currency: "USD",
+		locale:   "nl-NL",
+		entries: []Entry{
+			{
+				Date:        "2015-03-12",
+				Description: "Buy present",
+				Change:      123456,
+			},
+		},
+		expected: `
+Datum      | Omschrijving              | Verandering
+12-03-2015 | Buy present               |   $ 1.234,56 
+`,
+	},
+	{
+		name:     "Dutch negative number with 3 digits before decimal point",
+		currency: "USD",
+		locale:   "nl-NL",
+		entries: []Entry{
+			{
+				Date:        "2015-03-12",
+				Description: "Buy present",
+				Change:      -12345,
+			},
+		},
+		expected: `
+Datum      | Omschrijving              | Verandering
+12-03-2015 | Buy present               |     $ 123,45-
+`,
+	},
+	{
+		name:     "American negative number with 3 digits before decimal point",
+		currency: "USD",
+		locale:   "en-US",
+		entries: []Entry{
+			{
+				Date:        "2015-03-12",
+				Description: "Buy present",
+				Change:      -12345,
+			},
+		},
+		expected: `
+Date       | Description               | Change
+03/12/2015 | Buy present               |     ($123.45)
+`,
+	},
+}
+
+var failTestCases = []struct {
+	name     string
+	currency string
+	locale   string
+	entries  []Entry
+}{
+	{
+		name:     "empty currency",
+		currency: "",
+		locale:   "en-US",
+		entries:  nil,
+	},
+	{
+		name:     "invalid currency",
+		currency: "ABC",
+		locale:   "en-US",
+		entries:  nil,
+	},
+	{
+		name:     "empty locale",
+		currency: "USD",
+		locale:   "",
+		entries:  nil,
+	},
+	{
+		name:     "invalid locale",
+		currency: "USD",
+		locale:   "nl-US",
+		entries:  nil,
+	},
+	{
+		name:     "invalid date (way too high month)",
+		currency: "USD",
+		locale:   "en-US",
+		entries: []Entry{
+			{
+				Date:        "2015-131-11",
+				Description: "Buy present",
+				Change:      12345,
+			},
+		},
+	},
+	{
+		name:     "invalid date (wrong separator)",
+		currency: "USD",
+		locale:   "en-US",
+		entries: []Entry{
+			{
+				Date:        "2015-12/11",
+				Description: "Buy present",
+				Change:      12345,
+			},
+		},
+	},
+}
+
+func TestFormatLedgerSuccess(t *testing.T) {
+	if TestVersion != testVersion {
+		t.Fatalf("Found TestVersion = %v, want %v", TestVersion, testVersion)
+	}
+	for _, tt := range successTestCases {
+		actual, err := FormatLedger(tt.currency, tt.locale, tt.entries)
+		// We don't expect errors for any of the test cases
+		if err != nil {
+			t.Fatalf("FormatLedger for input named %q returned error %q. Error not expected.",
+				tt.name, err)
+		}
+		expected := tt.expected[1:] // Strip initial newline
+		if actual != expected {
+			t.Fatalf("FormatLedger for input named %q was expected to return...\n%s\n...but returned...\n%s",
+				tt.name, strings.Replace(expected, " ", "_", -1), strings.Replace(actual, " ", "_", -1))
+		}
+	}
+}
+
+func TestFormatLedgerFail(t *testing.T) {
```

Comment:
This is super nitpicky to the point where maybe it's irrelevant, but I like it when there's a symmetry between the names of the tests: _success_ vs _failure_, or _succeed_ vs _fail_.

If I'm being too pedantic please say so, and I won't take it personally.